File: TwentyFortyEight/TFEAppDelegate.m
//
//  TFEAppDelegate.m
//  TwentyFortyEight
//
//  Created by Joshua Caswell on 2/28/16.
//  Copyright © 2016 Josh Caswell. All rights reserved.
//

#import "TFEAppDelegate.h"
#import "TFEMainScene.h"
#import "TFEGameController.h"

@implementation TFEAppDelegate

- (void)applicationDidFinishLaunching:(NSNotification *)notification
{
    TFEGameController * controller = [TFEGameController new];
    [self setGameController:controller];
    
    [[self mainWindow] setContentViewController:controller];
    [[self mainWindow] makeFirstResponder:controller];
    
    [[self mainWindow] makeKeyAndOrderFront:self];
}

@end

File: Mac/TFEAppDelegate.swift
//
//  TFEAppDelegate.swift
//  TwentyFortyEight
//
//  Created by Joshua Caswell on 10/1/16.
//  Copyright © 2016 Josh Caswell. All rights reserved.
//

import Cocoa
import SpriteKit

@NSApplicationMain
class TFEAppDelegate : NSObject, NSApplicationDelegate
{
    @IBOutlet weak var mainWindow: NSWindow!
    @IBOutlet weak var mainView: SKView!
    
    var gameController: TFEGameController!
    
    func applicationDidFinishLaunching(_: Notification)
    {
        let controller = TFEGameController()
        self.gameController = controller
        
        self.mainWindow.contentViewController = controller
        self.mainWindow.makeFirstResponder(controller)
        
        self.mainWindow.makeKeyAndOrderFront(self)
    }
}
